### PR TITLE
Argmin argmax

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented argmin and argmax function.
related to #1020 and #1092

### CODE REVIEW

File: tests/chainer_tests/functions_tests/math_tests/test_minmax.py [hunk 1]
```diff
@@ -341,4 +341,99 @@ def test_pos_neg_duplicate_axis(self):
...
+
+    def test_forward_cpu(self):
+        self.check_forward(self.x)
+
+    def test_forward_axis_cpu(self):
+        for i in range(self.x.ndim):
+            self.check_forward(self.x, axis=i)
+
+    def test_forward_negative_axis_cpu(self):
+        self.check_forward(self.x, axis=-1)
```
Comment (ir5) [inline tests/chainer_tests/functions_tests/math_tests/test_minmax.py]: How about testing all valid negative axes, not only -1? The same thing may hold for other negative axis tests below.

File: chainer/functions/math/minmax.py [hunk 2]
```diff
@@ -75,6 +77,56 @@ def _fwd(self, x, xp):
...
+            else:
+                type_check.expect(
+                    -self.axis - 1 < in_types[0].ndim,
+                )
+
+    def forward(self, x):
+        xp = cuda.get_array_module(*x)
+        return xp.asarray(self._fwd(x[0], xp)),
+
+    def backward(self, x, gy):
```
Comment (beam2d) [inline chainer/functions/math/minmax.py]: Do we need backward? I think this function is not differentiable, thus it should return None in backward, which is the default behavior of the base class `Function`.

File: chainer/functions/math/minmax.py [hunk 3]
```diff
@@ -75,6 +77,56 @@ def _fwd(self, x, xp):
         return xp.amin(x, axis=self.axis, keepdims=self.keepdims)
 
 
+class IndexSelectorBase(function.Function):
+    """Select index of an array element from a given axis."""
+
+    def __init__(self, axis=None, keepdims=False):
```
Comment (beam2d) [inline chainer/functions/math/minmax.py]: keepdims is not used.

File: chainer/functions/math/minmax.py (hunk 2 above)
Comment (unnonouno) [inline chainer/functions/math/minmax.py]: Oh, i see. Do we need to test backward computation in the unit test?

**Comment by beam2d** [comment]
LGTM!

**Pull request merged**